nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/CommentsMigrator.java

```diff
@@ -59,64 +74,166 @@
+                                                    .map(entry -> (String) entry.get(ECM_UUID))
+                                                    .collect(Collectors.toList());
+
+        if (!rootCommentsFolderIds.isEmpty()) {
+            String query = buildAllCommentsQuery(rootCommentsFolderIds, HIDDEN_FOLDER_TYPE);
+            List<String> timestampCommentFoldersIds = session.queryProjection(query, 0, 0)
+                                                             .stream()
+                                                             .map(entry -> (String) entry.get(ECM_UUID))
+                                                             .collect(Collectors.toList());
+            parentIds.addAll(rootCommentsFolderIds);
+            parentIds.addAll(timestampCommentFoldersIds);
+        }
+        return parentIds;
+    }
+
+    /**
+     * @since 11.1
+     */
+    protected void migrateSessionRelationToProperty(CoreSession session, MigrationContext migrationContext) {
         CommentService commentComponent = Framework.getService(CommentService.class);
         CommentServiceConfig commentServiceConfig = commentComponent.getConfig();
         if (commentServiceConfig != null) {
             RelationManager relationManager = Framework.getService(RelationManager.class);
             Graph graph = relationManager.getGraph(commentServiceConfig.graphName, session);
             List<Statement> statements = graph.getStatements();
-            checkShutdownRequested();
+            checkShutdownRequested(migrationContext);
 
-            processBatched(BATCH_SIZE, statements,
-                    statement -> migrateComments(session, relationManager, commentServiceConfig, statement),
-                    "Migrating comments");
-            reportProgress("Done", statements.size(), statements.size());
+            processBatched(migrationContext, BATCH_SIZE, statements,
+                    statement -> migrateCommentsFromRelationToProperty(session, relationManager, commentServiceConfig,
+                            statement),
+                    "Migrating comments from Relation to Property");
+            reportProgress("Done Migrating from Relation to Property", statements.size(), statements.size());
+        }
+    }
+
+    /**
+     * @since 11.1
+     */
+    protected void migrateSessionPropertyToSecured(CoreSession session, MigrationContext migrationContext) {
+        List<DocumentModel> comments = getUnsecuredComments(session);
+        PropertyCommentManager commentManager = (PropertyCommentManager) Framework.getService(CommentManager.class);
+        processBatched(migrationContext, BATCH_SIZE, comments,
+                comment -> migrateCommentsFromPropertyToSecured(session, commentManager, comment),
+                "Migrating comments from Property to Secured");
+
+        session.query(GET_COMMENTS_FOLDERS_QUERY).forEach((folder -> session.removeDocument(folder.getRef())));
```

Comment:
Seems un-useful.

Reply:
that for cleaning the Comments folders, which will be empty at the end of our migration.
and we take the proposal of efge to use `session.removeDocuments(documentsToRemove);`